From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Matti=20Lehtim=C3=A4ki?= <matti.lehtimaki@jolla.com>
Date: Thu, 2 Feb 2023 15:59:43 +0200
Subject: [PATCH] ld: Add LD_AS_NEEDED functionality

---
 ld/ldmain.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ld/ldmain.c b/ld/ldmain.c
index 9290a189b0d..5765444adf8 100644
--- a/ld/ldmain.c
+++ b/ld/ldmain.c
@@ -307,6 +307,8 @@ main (int argc, char **argv)
     yydebug = 1;
   }
 #endif
+  if (getenv ("LD_AS_NEEDED") && atoi(getenv ("LD_AS_NEEDED")) > 0)
+    input_flags.add_DT_NEEDED_for_regular = true;
 
   config.build_constructors = true;
   config.rpath_separator = ':';
